host/entry.go

```diff
@@ -0,0 +1,293 @@
+// Copyright 2018 The ZikiChomgo Authors. All rights reserved.  Use of this source
+// code is governed by a license that can be found in the License file.
+
+package host
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"zikichombo.org/sio/libsio"
+	"zikichombo.org/sound"
+	"zikichombo.org/sound/ops"
+	"zikichombo.org/sound/sample"
+)
+
+// Entry is the type of a connection to host sound I/O capabilities.
+//
+// Entries may either be high level (eg AudioFlinger) or low level (eg ALSA hw).
+//
+// Multiple entries may exist for a given host.  see Names() in the relevant
+// entry_{runtime.GOOS}.go file for details.
+type Entry interface {
+	// Name returns the name of the entry and should be a valid
+	// name for the host.
+	Name() string
+	// DefaultForm returns the default form.  In case one form is not
+	// implemented or uniform accross all supported functionality (input, output, device),
+	// output default should be returned.
```

Comment:
What is *output default*? Is it pre-defined in `sound`?

Reply:
I am trying to say that for audio i/o devices the defaults can differ.  For example, there can be playback in stereo default and capture in mono.  Instead of listing all defaults for all functionality in the host.Entry interface, I chose to just have one and say that if the implementer has to choose due to conflicts between defaults for different functionality, she should choose the best default for playback/output functionality.  This idea comes from the notion that playback is probably most common and people playing with capture/duplex seem more likely to specify (at least that has been my frame of mind when playing this stuff).